### ISSUE TITLE
Documenting Refresh: HTTP header

### ISSUE THREAD

**Issue by karlcow** [httpwg/http-extensions#138]
Maybe @reschke 

In [RFC7538](http://tools.ietf.org/html/rfc7538#section-4) there is a mention of `meta refresh` in HTML. I have seen instances online of a `Refresh` HTTP header. I haven't found any documentation about it in the past (or maybe I missed the appropriate RFC). I [met the header again yesterday](https://bugzilla.mozilla.org/show_bug.cgi?id=668275#c52) in an instance of gmail Web site.

When requesting
`→ http --print hH https://mail.google.com/ 'User-Agent:Mozilla/5.0 (Android 4.4.4; Mobile; rv:44.0) Gecko/44.0 Firefox/44.0' 'Accept-Language:en-US,en;q=0.5'`

```http
HTTP/1.1 200 OK
Alt-Svc: quic=":443"; ma=604800; v="30,29,28,27,26,25"
Alternate-Protocol: 443:quic,p=1
Cache-Control: private, max-age=604800
Content-Length: 306
Content-Type: application/xhtml+xml; charset=ISO-8859-1
Date: Thu, 28 Jan 2016 00:57:39 GMT
Expires: Thu, 28 Jan 2016 00:57:39 GMT
Refresh: 0;URL=https://mail.google.com/mail/
Server: GSE
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block
```

You might not be able to reproduce gmail has tons of versions depending on the location, devices, headers, etc.

I have started to [test a bit and write about it last year](http://www.otsukare.info/2015/03/26/refresh-http-header). It seems implemented uniformly, but that would require a bit more test. 

What would be the best course of actions for documenting it? Maybe I can give it a shot.

**Comment by phluid61** [comment]
Data point: http://www.otsukare.info/2015/03/26/refresh-http-header

**Comment by mnot** [comment]
A short Internet-Draft might do it. Would need to align with https://html.spec.whatwg.org/#attr-meta-http-equiv-refresh

@annevk thoughts?

**Comment by karlcow** [comment]
Using a very dumb query on [Google BigQuery](https://bigquery.cloud.google.com/) on httparchive data (Maybe someone can do something better than me) just to have a hunch on what was out there.

```sql
SELECT respOtherHeaders
FROM [httparchive:runs.latest_requests] 
WHERE REGEXP_MATCH(respOtherHeaders, r'Refresh =') 
LIMIT 1000
```

Then cleaning a bit the data being returned.

I see some patterns to be tested (not complete)

```
Refresh: 0
Refresh: 0.1; URL=http://example.com
Refresh: 0; http://example.com
Refresh: 0; url=/somewhere
Refresh: 0; url=http://example.com/somewhere
Refresh: 0;url=/somewhere
Refresh: 0;url=http://example.com/somewhere
Refresh: 1020
Refresh: 1; url=somewhere
Refresh: 1;/
Refresh: 1;url=?somewhere
Refresh: Every 15 Minutes
Refresh: true
```

o_0 `Every 15 Minutes`.

Anyway as usual sometimes a big mess but a subset will probably be interoperable.

**Comment by annevk** [comment]
I was thinking we'd just define this header in HTML directly at some point. Maybe @zcorpan is interested in doing that. He touched refresh last I believe.

**Comment by annevk** [comment]
(Mainly because defining this properly requires tight integration with the navigation algorithm.)

**Comment by reschke** [comment]
Defining it there would imply that it only applies to browsers. Is this the case?

**Comment by annevk** [comment]
I don't see how that would imply that, but I also have not heard about non-browser support.

**Comment by zcorpan** [comment]
There's a bug about this at https://www.w3.org/Bugs/Public/show_bug.cgi?id=28339

**Comment by mnot** [comment]
I don't think it particularly matters where it's defined, as long as we make sure it's reviewed by the appropriate folks and registered.

**Comment by mnot** [comment]
Closing issue. If someone decides to submit an Internet-Draft, we'll take a look. If it gets done in HTML, give us a shout and we'll review.

**Comment by buro9** [comment]
Just a note and not a re-open request:

The HTML meta refresh is a `http-equiv`, the existence of meta refresh in HTML is to declare that the HTTP refresh header also exists. Every browser appears to treat it that way, to implement the HTML meta refresh they have all implemented the HTTP refresh.

The bug as it stands is that there is no standard documentation of a refresh header in HTTP (outside of HTML documentation).

I have used the HTTP header at scale and the HTML spec for the header value format works as expected everywhere.

I have also used it in JSON APIs where again it worked (except with primitive bots, which was what I was aiming for that time). I am not convinced the "solve in the HTML spec" approach is the right one as it does work outside of HTML.
